src/sentry/integrations/slack/message_builder/image_block_builder.py

```diff
@@ -0,0 +1,81 @@
+import datetime
+import logging
+
+from sentry import charts, features
+from sentry.api import client
+from sentry.charts.types import ChartType
+from sentry.integrations.slack.message_builder import SlackBlock
+from sentry.integrations.slack.message_builder.base.block import BlockSlackMessageBuilder
+from sentry.integrations.slack.message_builder.time_utils import get_approx_start_time
+from sentry.issues.grouptype import PerformanceP95EndpointRegressionGroupType
+from sentry.models.apikey import ApiKey
+from sentry.models.group import Group
+from sentry.snuba.referrer import Referrer
+
+logger = logging.getLogger("sentry.integrations.slack")
+
+
+class ImageBlockBuilder(BlockSlackMessageBuilder):
+    def __init__(self, group: Group) -> None:
+        super().__init__()
+        self.group = group
+        self.event = group.get_recommended_event_for_environments()
+
+    def build(self) -> SlackBlock | None:
+        if (
+            features.has("organizations:slack-endpoint-regression-image", self.group.organization)
+            and self.group.issue_type == PerformanceP95EndpointRegressionGroupType
+        ):
+            return self._build_endpoint_regression_image_block()
+
+        # TODO: Add support for other issue alerts
+        logger.warning("build_issue_alert_image.not_supported")
+        return None
+
+    def _build_endpoint_regression_image_block(self) -> SlackBlock | None:
+        logger.info(
+            "build_endpoint_regression_image.attempt",
+            extra={
+                "group_id": self.group.id,
+            },
+        )
+        endpoint = "events-stats/"
+        organization = self.group.organization
+        try:
+            resp = client.get(
+                auth=ApiKey(organization_id=organization.id, scope_list=["org:read"]),
+                user=None,
+                path=f"/organizations/{organization.slug}/{endpoint}",
+                data={
+                    "yAxis": ["count()", "p95(transaction.duration)"],
+                    "referrer": Referrer.API_ALERTS_CHARTCUTERIE,
+                    "query": f"event.type:transaction transaction:{self.event.transaction}",
+                    "project": self.group.project.id,
+                    "start": get_approx_start_time(self.group),
+                    "end": datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
```

Comment:
Does this always fetch a time series from when the regression starts to now? If so, then there's 2 issues
1. It doesn't contain any data from before the regression so instead of showing a regression, it'd just show the regressed state.
2. If the regression started 90 days ago, you'd be querying for 90 days of data which is a lot and may timeout.
Can we do what the frontend does and take a 14 day period centered around the regression?

Reply:
There's no equivalent on the backend yet because there hasn't been a need for it. But I would do something like

```
"start": get_approx_start_time(self.group) - timedelta(days=7),
"end": datetime.datetime.now(),
```

So it gets a few days of history for the regression instead of only showing the regressed portion.

Take a simple regression like

```python
                 -----------
                /
               /
     ----------
time 0        10          20
```
Where the the regression starts at time 10, if you go from time 10 to now, you'd only show the latter half of the cart but miss the baseline portion of that chart that demonstrates the regression.